### DISCUSSION TITLE
System.Data.SqlClient flaged as dependency by SourceBuild, but not present in any project.assets.json files

### DISCUSSION THREAD

**Discussion by JanKrivanek** [dotnet/source-build#3499] [Q&A] [answered]
### Background
System.Data.SqlClient v4.8.3 is reported as dependency of MSBuild (https://devdiv.visualstudio.com/DevDiv/_componentGovernance/DotNet-msbuild-Trusted?_a=components&typeId=8799614&alerts-view-option=active)
Looking into a sample build run: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=7861261&view=results, th dependency shows only in the SourceBuild ([binlog](https://artprodscussu1.artifacts.visualstudio.com/A011b8bdf-6d56-4f87-be0d-0092136884d9/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2RldmRpdi9wcm9qZWN0SWQvMGJkYmM1OTAtYTA2Mi00YzNmLWIwZjYtOTM4M2Y2Nzg2NWVlL2J1aWxkSWQvNzg2MTI2MS9hcnRpZmFjdE5hbWUvQnVpbGRMb2dzX1NvdXJjZUJ1aWxkX01hbmFnZWRfQXR0ZW1wdDE1/content?format=file&subPath=%2Fartifacts%2Fsource-build%2Fself%2Fsrc%2Fartifacts%2Fsourcebuild.binlog))

<img width="603" alt="01" src="https://github.com/dotnet/source-build/assets/3809076/85e955ab-b0a9-45c9-8bbb-99993a7743be">

However it doesn't show in neither project.assets.json files that are part of the build:

<img width="184" alt="02" src="https://github.com/dotnet/source-build/assets/3809076/1fc3f566-56ca-4a80-81d5-0a9e71d1f51a">

### Where does this happen

MSBuild pipeline: https://devdiv.visualstudio.com/DevDiv/_build?definitionId=9434&_a=summary, build of main branch

### Possible Issue

While the dependency is part of the nuget cache, it doesn't seem to be pulled by the build. Rather I suspect it being pulled by some of the SourceBuild components/dependencies.
Is there a chance to make a snapshot before the build itself and then do a diff of those (in order to prevent accounting SourceBuild components as target build components)?

**Comment by ghost** [comment]
I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one area label.

**Comment by MichaelSimons** [comment, marked as answer]
The source of this packages comes from this dependency - https://github.com/dotnet/msbuild/blob/main/eng/Version.Details.xml#L4.  MSBuild is referencing a very old version of this and should be upgraded to the latest 8.0 version and this issue will be resolved.  I would recommend ensuring there is a darc subscription to keep this up-to-date.

The source-build-reference-packages repo contains all of the reference packages needed to build .NET from source.  The SB infrastructure pulls in these packages implicitly via the source build metadata on the Version.Details.xml dependencies.

**Comment by JanKrivanek** [comment]
Thank you @MichaelSimons - we had a subscription to .NET 7 Eng channel - so we updated it
